### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by xiangshen-dk**
Some libraries are out of date. This PR will also allow users to use JDK 8+ to build and deploy the app.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,7 +90,7 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <version>3.7.0</version>
```
Comment (ricalo) [inline pom.xml:77]: Interesting that this plugin got downgraded. Is this just what your IDE produced? Or did you manually downgrade this one?

File: README.md [hunk 2]
```diff
@@ -17,14 +17,16 @@ solution. You can find the sample code for the Android client code in the
 - Sign up for [Firebase](https://firebase.google.com/) and create a new project
   in the [Firebase console](https://console.firebase.google.com/).
 - Install the following tools in your development environment:
-  - [Python 2.7](https://www.python.org/downloads/)
```
Comment (ricalo) [inline README.md:20]: Happy to remove stuff that is not needed, but could you explain your rationale?

File: README.md [hunk 3]
```diff
@@ -17,14 +17,16 @@ solution. You can find the sample code for the Android client code in the
...
   - [Java 8](https://java.com/en/download/)
   - [Apache Maven](https://maven.apache.org/)
   - [Google Cloud SDK](https://cloud.google.com/sdk/)
 
 > **Note**: Firebase is a Google product, independent from Google Cloud
 > Platform.
 
+> App Engine Flexible Environment [needs Java 8 Runtime](https://cloud.google.com/appengine/docs/flexible/java/setting-up-environment). However, in your local development environment you can
```
Comment (ricalo) [inline README.md:27]: A few comments with this paragraph:

* It doesn't need to be formatted as a [quote][0], just a regular paragraph should work. This is especially important if there's another quote nearby, like in the previous paragraph.
* This reads as if *App Engine Flexible Environment needs the Java 8 runtime*, which it doesn't necessarily. App Engine Flexible Environment can use many runtimes, including Java 8.

The sample *needs* App Engine Flexible Environment with the Java 8 runtime. There might be a better way to phrase this. Could you help me understand what is your goal with this addition?

[0]: https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#quoting-text

File: pom.xml (hunk 1 above)
Comment (xiangshen-dk) [reply pom.xml:77]: Good catch! This is the default version used by my IDE's app engine plugin. I bumped it to 3.8.1 which is the latest and everything appears fine.

File: README.md (hunk 2 above)
Comment (xiangshen-dk) [reply README.md:20]: Python 2.7 is not a direct dependency for this project. It's required by the Cloud SDK, which supports both Python 2 and 3.
https://cloud.google.com/sdk/install

File: README.md (hunk 3 above)
Comment (xiangshen-dk) [reply README.md]: Updated the sentence based on your feedback. Thanks!

File: README.md [hunk 4]
```diff
@@ -17,14 +17,17 @@ solution. You can find the sample code for the Android client code in the
...
 - Install the following tools in your development environment:
-  - [Python 2.7](https://www.python.org/downloads/)
   - [Java 8](https://java.com/en/download/)
   - [Apache Maven](https://maven.apache.org/)
   - [Google Cloud SDK](https://cloud.google.com/sdk/)
 
 > **Note**: Firebase is a Google product, independent from Google Cloud
 > Platform.
 
+A Java application deploys to App Engine Flexible Environment [needs to use Java 8 Runtime](https://cloud.google.com/appengine/docs/flexible/java/setting-up-environment).
```
Comment (ricalo) [inline README.md:27]: deployed?

**Review by ricalo — CHANGES_REQUESTED** [review]
Just one more comment and I think we're done with this review.

**Review by ricalo — APPROVED** [review]

**Comment by xiangshen-dk** [comment]
Corrected the typo. Thanks for your review!

**Pull request merged**